test/Microsoft.AspNetCore.SignalR.Tests.Utils/PipeReaderExtensions.cs

```diff
@@ -51,6 +51,19 @@ public static class PipeReaderExtensions
             }
         }
 
+
+        public static async Task ReadAsync(this PipeReader pipeReader, int numBytes)
+        {
+            var result = await pipeReader.ReadAsync();
+            if (result.Buffer.Length < numBytes)
+            {
+                pipeReader.AdvanceTo(result.Buffer.Start, result.Buffer.End);
```

Comment:
This should loop until you're actually able to read the requested number of bytes.